src/raftstore/store/snap.rs

```diff
@@ -61,223 +117,517 @@ impl Display for SnapKey {
     }
 }
 
-/// Name prefix for the self-generated snapshot file.
-const SNAP_GEN_PREFIX: &'static str = "gen";
-/// Name prefix for the received snapshot file.
-const SNAP_REV_PREFIX: &'static str = "rev";
+pub trait SendSnapshotFileBuilder {
+    fn build(&mut self, snap: &DbSnapshot, region: &Region) -> raft::Result<()>;
```

Comment:
Yes, we should think a better name, Builder confuses me. Builder like a Factory, `build` should return some built object. Btw, the builder even has a `total_size` function, what does it mean for a Builder?